crates/l2/prover/zkvm/interface/src/lib.rs

```diff
@@ -100,32 +107,45 @@ pub mod trie {
             } else {
                 // Add or update AccountState in the trie
                 // Fetch current state or create a new state to be inserted
-                let mut account_state = match state_trie.get(&hashed_address)? {
-                    Some(encoded_state) => AccountState::decode(&encoded_state)?,
-                    None => AccountState::default(),
+                let account_state = state_trie.get(&hashed_address);
+
+                // if there isn't a path into the account (inconsistent tree error), then
+                // it's potentially a new account. This is because we're using pruned tries
+                // so the path into a new account might not be included in the pruned state trie.
+                let mut account_state = match account_state {
+                    Ok(Some(encoded_state)) => AccountState::decode(&encoded_state)?,
```

Comment:
Can't we use "?" instead of matching with multiple Oks/Err?

Reply:
We can't because we want to match the `InconsistentTree` error and interpret it as a "state not found" error instead of propagating it